### PR TITLE
Added Missing Data Modeling Helpers

### PR DESCRIPTION
**Pull request by doctrino**
## Description
Made all test resources for data modeling automatically created, instead of using a script. 

Found a few bug along the way, and added missing helper methods. 

## Checklist:
- [x] Tests added/updated.
- [x] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [x] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [x] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: tests/tests_integration/test_api/test_data_modeling/test_instances.py [hunk 1]
```diff
@@ -24,62 +27,59 @@
...
 class TestInstancesAPI:
-    def test_list_nodes(self, cognite_client: CogniteClient, cdf_nodes: NodeList) -> None:
+    def test_list_nodes(self, cognite_client: CogniteClient, movie_nodes: NodeList) -> None:
         # Act
-        actual_nodes = cognite_client.data_modeling.instances.list(limit=-1)
+        listed_nodes = cognite_client.data_modeling.instances.list(limit=-1, instance_type="node")
 
         # Assert
-        assert sorted(actual_nodes, key=lambda v: v.external_id) == sorted(cdf_nodes, key=lambda v: v.external_id)
+        assert set(movie_nodes.as_ids()) <= set(listed_nodes.as_ids())
```
Comment (haakonvt) [inline tests/tests_integration/test_api/test_data_modeling/test_instances.py:75]: Should we guard against empty sets passing the assert?

File: tests/tests_integration/test_api/test_data_modeling/test_spaces.py [hunk 2]
```diff
@@ -10,8 +10,10 @@
 from cognite.client.exceptions import CogniteAPIError
 
 
-@pytest.fixture(scope="function")
-def cdf_spaces(cognite_client: CogniteClient) -> SpaceList:
+@pytest.fixture(scope="session")
+def cdf_spaces(cognite_client: CogniteClient, integration_test_space: Space) -> SpaceList:
+    # The integration test space is created in the fixture integration_test_space.
+    # This ensures that at least one space exists in CDF.
```
Comment (haakonvt) [inline tests/tests_integration/test_api/test_data_modeling/test_spaces.py:16]: I prefer to use `usefixtures` to indicate fixture dependency when not directly needed:

```suggestion
@pytest.fixture(scope="session")
@pytest.mark.usefixtures("integration_test_space")
def cdf_spaces(cognite_client: CogniteClient) -> SpaceList:
```

File: tests/tests_integration/test_api/test_data_modeling/test_spaces.py [hunk 3]
```diff
@@ -32,9 +34,9 @@ class TestSpacesAPI:
     def test_list(self, cognite_client: CogniteClient, cdf_spaces: SpaceList) -> None:
         actual_space_in_cdf = cognite_client.data_modeling.spaces.list(limit=-1)
 
-        assert _dump(actual_space_in_cdf) == _dump(cdf_spaces)
+        assert actual_space_in_cdf.as_apply_list() == cdf_spaces.as_apply_list()
 
-    def test_list_include_global(self, cognite_client: CogniteClient) -> None:
+    def test_list_include_global(self, cognite_client: CogniteClient, integration_test_space: Space) -> None:
```
Comment (haakonvt) [inline tests/tests_integration/test_api/test_data_modeling/test_spaces.py:39]:
```suggestion
    @pytest.mark.usefixtures("integration_test_space")
    def test_list_include_global(self, cognite_client: CogniteClient) -> None:
```

**Review by haakonvt — COMMENTED** [review]
A handful comments

File: cognite/client/data_classes/data_modeling/ids.py [hunk 4]
```diff
@@ -105,14 +105,14 @@ def instance_type(self) -> Literal["node", "edge"]:
 T_InstanceId = TypeVar("T_InstanceId", bound=InstanceId)
 
 
-@dataclass
+@dataclass(frozen=True)
 class NodeId(InstanceId):
-    _instance_type = "node"  # type: ignore[assignment]
+    _instance_type: ClassVar[Literal["node", "edge"]] = "node"
```
[the remark this answers is not in the document]
Comment (doctrino) [reply cognite/client/data_classes/data_modeling/ids.py:110]: If we add a comment for that, we would have to add many comments in the code base, as this trick is used all over in filtering, and aggregation. As this field is private I lean towards not adding an extra comment for it.

File: tests/tests_integration/test_api/test_data_modeling/test_spaces.py (hunk 2 above)
Comment (doctrino) [reply tests/tests_integration/test_api/test_data_modeling/test_spaces.py:16]: I learned something new, did not know about the usefixture fixture.

File: cognite/client/_api/data_modeling/graphql.py [hunk 5]
```diff
@@ -14,7 +14,9 @@ class DataModelingGraphQLAPI(APIClient):
     def _post_graphql(self, url_path: str, json: dict) -> dict[str, Any]:
         res = self._post(url_path=url_path, json=json).json()
         # Errors can be passed both at top level and nested in the response:
-        errors = res.get("errors", []) + ((res["data"].get("upsertGraphQlDmlVersion") or {}).get("errors") or [])
+        errors = res.get("errors", []) + (
+            (res.get("data", {}).get("upsertGraphQlDmlVersion") or {}).get("errors") or []
```
[the remark this answers is not in the document]
Comment (doctrino) [reply cognite/client/_api/data_modeling/graphql.py:18]: Yes, first thing I did when I set up the fixture for auto deploying the data model used for testing. 

I have taken my own advice an added a failing test to demonstrate how I triggered it.

**Review by haakonvt — APPROVED** [review]
LGTM, (failing tests looks unrelated)

**Pull request merged**